### DISCUSSION TITLE
Datagrid row actions slow

### DISCUSSION THREAD

**Discussion by davazlan** [blazority/support#62] [Q&A]
The datagrid row actions pop up quite slowly. What could be impacting the performance?
My implementation looks like this currently:
```
<DatagridRowActions TRowData="Model" CanShow="@CanShowActions">
                    <DatagridRowActionItem>
                        <Icon Shape="note" Status="Status.Info"></Icon>
                        <Button Variant="Info" Flat="true" Small="true" type="button" OnClick="() => EditModel(context)">Edit</Button>
                    </DatagridRowActionItem>
                    <DatagridRowActionItem>
                        <Icon Shape="eraser" Status="Status.Danger"></Icon>
                        <Button Variant="Danger" Flat="true" Small="true" type="button" OnClick="() => DeleteModel(context)">Delete</Button>
                    </DatagridRowActionItem>
                </DatagridRowActions>
```

Could it be the button making them render slowly?

**Comment by davazlan** [comment]
So... I changed my action items. Now I have:
```
<DatagridRowActions TRowData="Model" CanShow="@CanShowActions">
   <DatagridActionItem>
      <span @onclick="() => Edit(context)">
         <Icon Shape="note" Status="Status.Info"/>
         Edit
      </span>
   </DatagridActionItem>
</DatagridRowActions>
```
And the action menu is still quite slow. Is there a way to make the lag shorter? The user clicks on the three dots (action cell) and has to wait for 2-3 seconds before the menu actually pops up.

**Reply by karan-kang** [reply]
I tried to reproduce this in a test app with both Blazor server and WASM and unfortunately was not able to reproduce this issue. You can even see this working as expected in the demo.

If you are using Blazor server, I would suggest looking at any other rendering delays.
